### DISCUSSION TITLE
Flux manager barfs parsing YAML manifests encoded with anything besides UTF-8

### DISCUSSION THREAD

**Discussion by danports** [renovatebot/renovate#23599] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest, from GitLab project template

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Probably never worked, but I didn't notice it until I reviewed my Renovate logs to look into something else.

### Describe the problem

js-yaml throws `YAMLException: null byte is not allowed in input (1:4)` and the Flux manager logs the error (`Failed to parse Flux manifest`). It seems like this happens with YAML files encoded with anything other than UTF-8 (e.g. UTF-16LE where, of course, there are a lot of null bytes), because we [hard code the UTF-8 encoding](https://github.com/renovatebot/renovate/blob/680f74395a02b5737b61e4342dd67f667c02afdc/lib/modules/manager/flux/extract.ts#L304C32-L304C32). I'm not sure if `fs.readFile` can detect the encoding if it is not specified. I would think this would be a problem for other managers too, but maybe encodings besides UTF-8 are relatively rare so we don't care?

### Have you created a minimal reproduction repository?

TODO

**Comment by viceice** [comment]
yes, we currently don't care and it's very difficult to detect right encoding

**Reply by danports** [reply]
I see - that's too bad. I've worked with other languages/frameworks that can detect file encodings automatically, but from searching around a bit it doesn't seem that Node has something comparable, at least not without adding another dependency. 😞 I guess I can reencode the problematic files to work around this, but it's not a big deal since there aren't any dependencies to extract from the files in question (encrypted sealed-secrets manifests) anyway.

**Reply by viceice** [reply]
maybe just exclude / disable those files
